src/mbgl/util/http_timeout.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <mbgl/storage/response.hpp>
+#include <mbgl/util/optional.hpp>
+#include <mbgl/util/chrono.hpp>
+
+namespace mbgl {
+namespace http {
+
+static Duration errorRetryTimeout(Response::Error::Reason failedRequestReason, uint32_t failedRequests, optional<Timestamp> retryAfter = {}) {
```

Comment:
Using an internal header for the method declaration is 👍 , but the definition should go in a sibling http_timeout.cpp file. And then remove `static` here.